firefox-ios/Storage/Rust/RustLogins.swift

```diff
@@ -1056,17 +827,15 @@ public class RustLogins: LoginsProtocol {
         }
     }
 
-    private func getKeychainData(rustKeys: RustLoginEncryptionKeys, completion: @escaping (String?, String?) -> Void) {
-        DispatchQueue.global(qos: .background).sync {
-            let key = rustKeys.keychain.string(forKey: rustKeys.loginPerFieldKeychainKey)
-            let encryptedCanaryPhrase = rustKeys.keychain.string(forKey: rustKeys.canaryPhraseKey)
-            completion(key, encryptedCanaryPhrase)
-        }
+    private func getKeychainData(rustKeys: RustLoginEncryptionKeys) -> (String?, String?) {
+        let key = rustKeys.keychain.string(forKey: rustKeys.loginPerFieldKeychainKey)
+        let encryptedCanaryPhrase = rustKeys.keychain.string(forKey: rustKeys.canaryPhraseKey)
+        return (key, encryptedCanaryPhrase)
```

Comment:
no need for background call here?

Reply:
I had some code before that called `getKeychainData` in the background and removed it 😓 Looking at it now, I think it makes sense to keep it in the background still, since I am not sure how compute intensive it is. Good catch !